newm-server/src/main/kotlin/io/newm/server/config/repo/ConfigRepositoryImpl.kt

```diff
@@ -28,4 +28,11 @@ internal class ConfigRepositoryImpl : ConfigRepository {
     override suspend fun getBoolean(id: String): Boolean = getString(id).toBoolean()
 
     override suspend fun getBooleans(id: String): List<Boolean> = getStrings(id).map(String::toBoolean)
+
+    override suspend fun putString(id: String, value: String) = transaction {
+        ConfigEntity.new(id) {
+            this.value = value
+        }
+        Unit
+    }
```

Comment:
`.new`  fails if already exists. What do you think of somthing like this?
```
        ConfigEntity.findById(id)?.let {
            it.value = value
        } ?: ConfigEntity.new {
            this.value = value
        }
```

Reply:
Also, one way to avoid the weird `Unit` at the end, is to explicitly add the `Unit` to the signature:
```
override suspend fun putString(id: String, value: String): Unit
```

Reply:
We actually want `.new` to fail if a key already exists. We don't want code arbitrarily overwriting our salt or password config values. The code that calls this already has a check to ensure the key does not yet exist and will fail early.

So, this code is intended to do a one-time-only set of the values. After that, the only way to override it is to manually go into the config db and change something.

The way I see this working is we set up the server, give one user admin rights who sets the salt/pwd over an https connection, then take away all admin rights.